app/controllers/owned_tag_sets_controller.rb

```diff
@@ -144,7 +144,17 @@ def new
 
   def create
     @tag_set = OwnedTagSet.new(owned_tag_set_params)
-    @tag_set.add_owner(current_user.default_pseud)
+
+    # Check if the current user already owns the tag set, and only add them if they don't.
+    #
+    # This happens when the user adds themselves explicitly as an owner.
+    # This check is done by direct Enumerable.find rather than ActiveRecord `find_by`
+    # as we cannot use `find_by` before the record is stored in the database with `.save`
+    current_user_ownership = @tag_set.tag_set_ownerships.find do |ownership|
+      ownership.owner == true && ownership.pseud_id == current_user.default_pseud.id
+    end
+    @tag_set.add_owner(current_user.default_pseud) unless current_user_ownership
```

Comment:
I'm thinking I'd prefer this to live in the model somewhere, but bouncing some ideas around right now (with a few clarifications I think the Jira story could use) so feel free to wait on any other changes until I have an answer for you on that

Reply:
Ok, I think we can do this with a model validation in https://github.com/otwcode/otwarchive/blob/1bdb4441e193ca269bb31f8c4a78ab5db7d1b663/app/models/tagset_models/tag_set_ownership.rb:
```rb
class TagSetOwnership < ApplicationRecord
  belongs_to :pseud
  belongs_to :owned_tag_set

  validates :pseud, uniqueness: { scope: :owned_tag_set_id }
end
```
(credit to @Bilka2 for the above). This will validate that the pseud is unique and only appears for moderator or owner, not both. The message can be taken care of with Rails I18n, similar to https://github.com/otwcode/otwarchive/blob/dd6ec55a93d549cef4e9a43269dc1cebb975dd91/config/locales/models/en.yml#L109 (docs: https://github.com/otwcode/otwarchive/wiki/Internationalization-%28i18n%29-Standards#overriding-a-default-error-message); please let me know if additional examples would be helpful there.

With that validation in place, we won't need to worry about removing owners multiple times. So, instead of special-casing in the model, could you write a one-time Rake task to find and fix any duplicate owners?

Sorry for the wall of text, please don't hesitate to let us know if you have any questions!